src/core/loss_detection.c

```diff
@@ -456,6 +463,32 @@ QuicLossDetectionOnPacketSent(
             &Connection->CongestionControl, SentPacket->PacketLength);
     }
 
+    SentPacket->Flags.IsAppLimited = QuicCongestionControlIsAppLimited(&Connection->CongestionControl);
+
+    uint64_t SendPostedBytes = Connection->SendBuffer.PostedBytes;
+    BOOLEAN LossBufferEmpty = !LossDetection->LostPackets &&
```

Comment:
It seems like you are checking for "!LossDetection->LostPackets" to satisfy condition 4 in "Tracking application-limited phases" here: https://datatracker.ietf.org/doc/html/draft-cheng-iccrg-delivery-rate-estimation-02#section-2.3

"4.  All the packets considered lost have been retransmitted."

But this queue can still have packets in it after we've already retransmitted them. See QuicLossDetectionDetectAndHandleLostPackets, where we move packets to the LostPackets list after calling QuicLossDetectionRetransmitFrames to retransmit them.

Reply:
Agreed, it is a mistake. Any suggestion for the fix?

Reply:
I think we can check if QuicStreamCanSendNow returns TRUE for any stream. If it does, then we aren't application-limited. In addition to checking for unretransmitted lost packets, QuicStreamCanSendNow also includes a check for flow-control-limited situations (where we are limited by the peer application rather than the local application). I don't see flow-control-limited situations included in the rate estimation IETF draft, but I think it's appropriate to include them. I think this calls for followup with the BBR designers if it hasn't already been discussed.

Reply:
I just made a fix for the issue, but I don't have 100% confidence for that.
Plz have a double check.

Reply:
You're now following my advice to call QuicStreamCanSendNow, but QuicStreamCanSendNow includes a check for FC-blocked and per the discussion on the google group, FC-blocked should not be included in the set of criteria for calling the connection "app-limited". So this needs to be fixed to exclude that check.

Reply:
It's kind of ugly, but you could just add a check of QuicStreamAllowedByPeer(Stream) when setting the LossBufferEmpty variable.

Also, "LossBufferEmpty" is a very confusing name. You don't need a variable at all here- just add the logic to the set of checks in the "if" statement for calling QuicCongestionControlSetAppLimited.

Reply:
Tried to fix in the latest commit. Need double check from @maolson-msft